Add initial Oomph setup for m2e

HannesWell opened the pull request:
This PR adds an Oomph setup for m2e as part of issue #120 

At the moment the setup only contains the basic tasks based on [CONTRIBUTING.md](https://github.com/eclipse-m2e/m2e-core/blob/master/CONTRIBUTING.md)
- clone the m2e repo
- install all required m2e features (for dogfooding the latest snapshots are used)
- install those m2e connectors I could identify to be necessary (if there is a more elegant way please let me know)
- Import projects directly nested in m2e-core, m2e-maven-runtime, and m2e-core-tests and assign them to corresponding working-sets
- load the defined target-platform and perform refresh, clean, build
- Run an install build for m2e-maven-runtime
- set a JRE for JavaSE-11 to be used

Besides that more user preferences could be set (like activating save-actions) or more launch configs could be add (e.g. one launcher for an ordinary mvn clean verify build and one to build all, like the buildall,sh just startable from the IDE on Windows).

mickaelistria commented:
Can you please include into this PR the changes to CONTRIBUTING.md with instructions about how to use Oomph to provision the IDE?

laeubi reviewed (commented):
I think a launch config that launches an Eclipse-Dev-IDE would be handy.

HannesWell commented:
> Can you please include into this PR the changes to CONTRIBUTING.md with instructions about how to use Oomph to provision the IDE?

Sure. Am I correct in assuming that the m2e-setup will be included into the `Eclipse projects` catalog of Oomph? Otherwise it had to be imported as user-project.

If yes I can also submit a bug against Oomph as described in the [Eclipse Oomph Authoring article](https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Contributing_to_a_Project_Catalog) in order to initiate its inclusion.

HannesWell commented:
> I think a launch config that launches an Eclipse-Dev-IDE would be handy.

Sure. But for this it could be better to use the running Eclipse as Target-Platform or do you think the features/plugins in the dev-target-platform are sufficient for such an Eclipse?

laeubi commented:
The target should always be sufficient to run otherwise its not really a target :-)
But you can always add the current eclipse install as a target by adding 
`<location path="${eclipse_home}" type="Profile"/>`
to the target

HannesWell commented:
I've added the Launch-Config for an Eclipse-IDE and more launch configs for the different builds and updated the CONTRIBUTION.md mentioning the Oomph-setup and the launch-configs for build and IDE.

laeubi reviewed (approved):
I think this is good enough as a first step.

laeubi commented:
@HannesWell this currently has failing test, I'll restart the build to see if it is an intermediate failure, probably you need to rebase on master if that dosen't help.

HannesWell commented:
> I think this is good enough as a first step.

Agree. I have some more things to suggest, but as initial setup this should be sufficient. Especially because the setup should be included into Eclipse-project container. I think that should be done before we are going to fine-tune the setup.

keilw commented:
@HannesWell Which Eclipse versions are supported this way? And for at least the last 12 months or whatever  you can cover, are there or will there be "Available Updates" (I just spotted **m2e 1.17.2.20210219**) in the Dev IDE to fix those problems?

HannesWell commented:
> @HannesWell Which Eclipse versions are supported this way? And for at least the last 12 months or whatever you can cover, are there or will there be "Available Updates" (I just spotted **m2e 1.17.2.20210219**) in the Dev IDE to fix those problems?

Do you mean which Eclipse version is required to use this Oomph setup? It should be usable with older versions of Eclipse as long as the used Oomph Tasks are supported by the corresponding version of Eclipse-Oomph. But with that restriction I think you could use quite old versions of Eclipse.

> Actually **m2e 1.17.2.20210219** is useless because it would force an upgrade to Java 9 or 11 and we are bound to Java 8 in our environment. I have to experiment, if we could at least try 9 for the development environment while keeping the JRE at 8, hope all the other necessary plugins do not force e.g. using Java 11 because that would be a total no-go at this point?

The Eclipse platform requires Java-11 since version 4.17 (IIRC) which corresponds to Eclipse 2020-09. So yes the required plugins enforce Java-11 and I'm pretty sure it wont be possible to run Eclipse 2020-09 or later with an Java-9 JRE.
But of course you can use a Java-11 JRE to run Eclipse and develop against Java-8. Some Eclipse-products provided by Oomph are already shipped with a suitable JRE (see project JustJ/Oomph), so you don't have to install one on your system.

akurtakov commented:
Should we push this one too? Improvements can continue after that

laeubi commented:
Yes please if your are also fine with that go ahead.

keilw commented:
We luckily found a workaround for our Eclipse 2020-06 environments installing some of those plugins differently (also without Ooph there I assume ;-) I also saw JCenter would stay, but I guess the problem was because of a Bintray hosting and those won't come back.

akurtakov commented:
@HannesWell do you have other work pending? If not I would like to push the current patches

HannesWell commented:
> @HannesWell do you have other work pending? If not I would like to push the current patches

Not at the moment. There are more things that could be done in the setup, but I fully agree that this is fine as a first initial setup.

After this is merged I suggest to open a Bug report against Oomph to ask for inclusion of the m2e setup into the Eclipse-project container, so m2e appears on the Oomph Installer's Project selections page. I think this should be done before we talk about further improvements of the setup because then more people can already use the setup.
I can take care of this at the beginning of next week (today I'm busy with my every-day job).

akurtakov commented:
Thanks Hannes.
